### ISSUE TITLE
BDS: Server crash due to CompoundTag

### ISSUE THREAD

**Issue by ghost** [LiteLDev/LiteLoader#832]
### Exceptions module

Core

### Operating System

Windows 10

### LiteLoader version

2.7.0 b07ff5e

### BDS version

1.19.30

### What happened?

I ran into the problem of sudden crashes that can occur at any time.

_I think the problem is due to changing the NBT of the mob, player or item_

### Steps to reproduce?

1. Enchant item across Enchant Table
2. Fly away from the enchantment table by 20-30 blocks
3. Click the tool on the block and wait for the crash

**Another step:**
1. Find any village
2. Wait for the crash

### Relevant log output

```shell
[Crashed!]
-- Unhandled Exception in -> bedrock_server_mod.exe
-- Exception Code: 0xC0000005
-- C++ STL Exception detected!

[StackTrace] Function CompoundTag::deepCopy at 0x7FF7C3AA91C0  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C3AA91F3  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function CompoundTag::clone at 0x7FF7C3AA8FB0  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C3AA9014  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function Actor::getPersistingTradeOffers at 0x7FF7C40685A0  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C40685B5  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function TransformationComponent::maintainOldData at 0x7FF7C3E477C0  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C3E47B41  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function Translation::loadImpl at 0x7FF9FC027660  [LiteLoader.dll<2.7.0.0 PRIVATEBUILD>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF9FC02E9C0  [LiteLoader.dll<2.7.0.0 PRIVATEBUILD>]
[StackTrace] Function TransformationComponent::transformIfAble at 0x7FF7C3E483F0  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C3E48772  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function Actor::reloadComponents at 0x7FF7C4076D60  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C4076E8C  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function Actor::_updateComposition at 0x7FF7C405D930  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C405D9FD  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function Actor::reload at 0x7FF7C40767D0  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C4076BE7  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function Actor::tick at 0x7FF7C407F3C0  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C407F80B  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function ActorLegacyTickSystem::tickActorLegacyTickSystem at 0x7FF7C3B25110  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C3B25209  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function EntityRegistryBase::View<1,EntityContext,EntityRegistry,entt::type_list<FlagComponent<ActorInvalidRegionFlag>,FlagComponent<PlayerComponentFlag> >,ActorTickNeededComponent,FlagComponent<ActorFlag> >::each<void (__cdecl*)(ViewedEntityContextT<EntityContext,ActorTickNeededComponent,FlagComponent<ActorFlag> > & __ptr64)> at 0x7FF7C3B241B0  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C3B24451  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function ActorLegacyTickSystem::tick at 0x7FF7C3B24F60  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C3B25012  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function std::_Func_impl_no_alloc<<lambda_de46fc0e395ea1d73f8119238160f27c>,void,ITickingSystem &,TickingSystemId>::_Do_call at 0x7FF7C3AA0010  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C3AA0029  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function `anonymous namespace'::DefaultEntitySystemsCollection::foreachTickingSystem at 0x7FF7C3F0FDF0  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C3F10117  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function EntitySystems::tick at 0x7FF7C3AA08A0  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C3AA0943  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function <lambda_b44508b61f37969cf480280e06e54415>::operator() at 0x7FF7C4993DB0  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C4993F04  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function Level::tickEntitySystems at 0x7FF7C49A8B70  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C49A8BAA  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function Level::tick at 0x7FF7C49A7BE0  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C49A80D5  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function ServerLevel::tick at 0x7FF7C41B1AC0  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C41B1B0A  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function ???????? at 0x7FF9FB6F775E  [LiteLoader.Js.dll<2.7.0.0 PRIVATEBUILD>]
[StackTrace] Function ???????? at 0x7FF9FADEA890  [LiteLoader.Lua.dll<2.7.0.0 PRIVATEBUILD>]
[StackTrace] Function Schedule::nextTick at 0x7FF9FC074B20  [LiteLoader.dll<2.7.0.0 PRIVATEBUILD>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF9FC074D2A  [LiteLoader.dll<2.7.0.0 PRIVATEBUILD>]
[StackTrace] Function GameSession::tick at 0x7FF7C42C0B70  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C42C0BA8  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function Minecraft::update at 0x7FF7C3F9C660  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C3F9CE40  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function ServerInstance::_update at 0x7FF7C3FBAEA0  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C3FBAF9D  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function <lambda_9c72527c89bc5df41fe482e4153a365f>::operator() at 0x7FF7C3FB9630  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C3FB9A8E  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function std::thread::_Invoke<std::tuple<<lambda_9c72527c89bc5df41fe482e4153a365f> >,0> at 0x7FF7C3FB6930  [bedrock_server_mod.exe<1.19.30.04>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FF7C3FB693E  [bedrock_server_mod.exe<1.19.30.04>]
[StackTrace] Function configthreadlocale at 0x7FFA44771B20  [ucrtbase.dll<6.2.19041.789>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FFA44771BB2  [ucrtbase.dll<6.2.19041.789>]
[StackTrace] Function BaseThreadInitThunk at 0x7FFA45DF7020  [KERNEL32.DLL<6.2.19041.1889>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FFA45DF7034  [KERNEL32.DLL<6.2.19041.1889>]
[StackTrace] Function RtlUserThreadStart at 0x7FFA469A2680  [ntdll.dll<6.2.19041.1949>]
-- At File (null) : Line 0 
[StackTrace] Function ???????? at 0x7FFA469A26A1  [ntdll.dll<6.2.19041.1949>]
```

### Plugin list

```shell
16:22:54 INFO [Server] Plugin List [2]
16:22:54 INFO [Server] - LLMoney [v2.6.0] (LLMoney.dll)
16:22:54 INFO [Server]   EconomyCore for LiteLoaderBDS
16:22:54 INFO [Server] - PermissionAPI [v2.7.0] (PermissionAPI.dll)
16:22:54 INFO [Server]   Builtin & Powerful permission API for LiteLoaderBDS
16:22:54 INFO [Server]
16:22:54 INFO [Server]
16:22:54 INFO [Server] * Send command "ll list <Plugin Name>" for more information
```

**Comment by ShrBox** [comment]
#829

**Comment by dreamguxiang** [comment]
Fixed 225dfb6e05eb97e3a945829aa4759fc8a188afe5

**Comment by Jaciezyt** [comment]
Duplicate of #829
